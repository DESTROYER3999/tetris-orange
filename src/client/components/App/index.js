import { connect } from "react-redux";
import { hot } from "react-hot-loader";
import { setConfig } from "react-hot-loader"; //to remove
import "react-toastify/dist/ReactToastify.css";
import AppSub from "./subcomponent";

/*

setConfig must be present for hot-loader to work

*/

setConfig({
  reloadHooks: false
});

const mapStateToProps = state => ({
  playerName: state.player.name,
  roomName: state.room.name
});

export default connect(mapStateToProps)(hot(module)(AppSub));
