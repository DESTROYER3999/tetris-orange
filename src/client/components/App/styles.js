import styled from "styled-components";

export const StyledContainer = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  &::after {
    content: "";
    background: url(assets/tetris-theme.jpg);
    background-size: cover;
    opacity: 0.2;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    position: absolute;
    z-index: -1;
  }

  // background-size: cover;
`;
