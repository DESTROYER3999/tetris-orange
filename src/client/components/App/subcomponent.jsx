import React from "react";
import Lobby from "../Lobby";
import Header from "../Header";
import Game from "../Game";
import Login from "../Login";
import { StyledContainer } from "./styles";

const AppSub = ({ playerName, roomName }) => {
  return (
    <StyledContainer id="App">
      <Header />
      {!playerName ? <Login /> : !roomName ? <Lobby /> : <Game />}
    </StyledContainer>
  );
};

export default AppSub;
