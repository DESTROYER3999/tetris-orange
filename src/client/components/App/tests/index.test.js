import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import App from "../index";
import socket from "../../../services/socket-api";

const mockStore = configureStore();
const initialState = {
  player: { name: "" },
  room: { name: "" }
};

const store = mockStore(initialState);

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

describe("<App />", () => {
  test("App rendering with store without crashing", () => {
    const wrapper = shallow(<App store={store} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
