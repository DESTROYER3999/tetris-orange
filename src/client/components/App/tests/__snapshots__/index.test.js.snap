// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App rendering with store without crashing 1`] = `
<ContextProvider
  value={null}
>
  <AppSub
    dispatch={[Function]}
    playerName=""
    roomName=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
